OpenSim/Common/SegmentedQuinticBezierToolkit.cpp

```diff
@@ -66,13 +66,21 @@ void SegmentedQuinticBezierToolkit::
     datafile.close();
 } 
 
-void SegmentedQuinticBezierToolkit::
-    printBezierSplineFitCurves(const SimTK::Function_<double>& curveFit, 
-    SimTK::Matrix& ctrlPts, SimTK::Vector& xVal, SimTK::Vector& yVal, 
+void SegmentedQuinticBezierToolkit::printBezierSplineFitCurves(
+    const SimTK::Function_<double>& curveFit,
+    std::vector<SimTK::Vec6>& ctrlPtsX,
+    std::vector<SimTK::Vec6>& ctrlPtsY,
+    SimTK::Vector& xVal,
+    SimTK::Vector& yVal,
     std::string& filename)
 {
+
+    SimTK_ERRCHK_ALWAYS(ctrlPtsX.size() == ctrlPtsY.size(),
+        "SegmentedQuinticBezierToolkit::printBezierSplineFitCurves",
+        "Error: X and Y control points must have same number of elements");
+
         std::string caller = "printBezierSplineFitCurves";
-        int nbezier =  int(ctrlPts.ncol()/2.0);
+        int nbezier =  ctrlPtsX.size();
         int rows = NUM_SAMPLE_PTS*nbezier - (nbezier-1);
```

Comment:
if I understand correctly, a `ptrdiff` will overflow right? So that will still wreck it.
Perhaps a runtime check that throws for a zero length vector, and use `size_t`?

Reply:
`ptrdiff_t` is signed, so under-/over-flow is undefined

Reply:
If the original implementation wasn't throwing with zero points then yours should not either - throwing in order to avoid using the appropriate signed int (`ptrdiff_t`) isn't a good substitute imo.

Reply:
Thats fair.
Removed assertion, rewritten such that no negative numbers arise.